File: 2478-longest-nice-subarray/longest-nice-subarray.java
class Solution {
    public int longestNiceSubarray(int[] nums) {
        int left = 0, usedBits = 0, maxLen = 0;

        for (int right = 0; right < nums.length; right++) {
            while ((usedBits & nums[right]) != 0) {
                usedBits ^= nums[left];
                left++;
            }

            usedBits |= nums[right];

            maxLen = Math.max(maxLen, right - left + 1);
        }

        return maxLen;
    }
}

File: 2478-longest-nice-subarray/longest-nice-subarray.go
func longestNiceSubarray(nums []int) int {
    left := 0;
    usedBits := 0;
    maxLen := 0;

    for right := 0; right < len(nums); right++ {

        for (usedBits & nums[right]) != 0 {
            usedBits ^= nums[left];
            left++;
        }

        usedBits |= nums[right];

        windowLen := right - left + 1;

        if windowLen > maxLen {
            maxLen = windowLen;
        }
    }

    return maxLen;
}